src/core/ConfigManager.cpp

```diff
@@ -334,27 +334,23 @@ void ConfigManager::addRecentlyOpenedProject(const QString & file)
 
 
 
-const QString & ConfigManager::value(const QString & cls,
+const QString ConfigManager::value(const QString & cls,
 					const QString & attribute) const
 {
-	if(m_settings.contains(cls))
+	if (m_settings.contains(cls))
 	{
-		for(stringPairVector::const_iterator it =
-						m_settings[cls].begin();
-					it != m_settings[cls].end(); ++it)
+		for (const auto& setting : m_settings[cls]) 
 		{
-			if((*it).first == attribute)
+			if (setting.first == attribute)
 			{
-				return (*it).second ;
+				return setting.second;
 			}
 		}
 	}
-	static QString empty;
-	return empty;
```

Comment:
``ConfigManager::value`` was segfaulting given the ``QVector`` changes. Not exactly sure why, but I believe it had something to do with returning the empty static ``QString``, as to which I just returned "" instead.

Reply:
Do you have a stack trace for the crash without this change? That might help finding the real issue.

Reply:
Also, the 3-parameter overload may return a reference to a local variable after this change, as spotted by CI builds.

Reply:
That sounds more like a dangling reference problem. It's a good idea to return a `std::optional` later, BTW.

Reply:
I found that [`QMap::operator[] const`](https://doc.qt.io/qt-5/qmap.html#operator-5b-5d-1) returns by value. For `QVector` which uses copy-on-write, this isn't a problem. For `std::vector`, however, it results in a dangling reference when a reference to its element is returned.

For future references, `std::map::operator[]` has no `const` version since `operator[]` may insert a default-constructed value for nonexistent keys. It's possible to use `std::map::at` which throws when the key is not found, though.